### PR TITLE
Fix ANIMATESHEET for custom particle effect

### PR DESCRIPTION
**Pull request by sebavan**
https://forum.babylonjs.com/t/cpu-particlesystem-with-custom-fragmentshader/38090/4

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/13533/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13533/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13533/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/13533/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/13533/merge
https://gui.babylonjs.com/?snapshot=refs/pull/13533/merge
https://nme.babylonjs.com/?snapshot=refs/pull/13533/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/13533/merge#BCU1XR#0

File: packages/dev/core/src/Particles/particleSystemComponent.ts [hunk 1]
```diff
@@ -93,6 +93,12 @@ Engine.prototype.createEffectForParticles = function (
         defines += "\n#define BILLBOARD\n";
     }
 
+    if (particleSystem?.isAnimationSheetEnabled) {
+        if (defines.indexOf(" ANIMATESHEET") === -1) {
```
Comment (Popov72) [inline packages/dev/core/src/Particles/particleSystemComponent.ts:97]: Just to be sure: the leading space is intended?

**Review by Popov72 — APPROVED** [review]

File: packages/dev/core/src/Particles/particleSystemComponent.ts (hunk 1 above)
Comment (sebavan) [reply packages/dev/core/src/Particles/particleSystemComponent.ts:97]: yup it is to find #define ANIMATE... like it is done for billboard for consistency but good to ask !!!

**Pull request merged**